src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,112 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { FilterBySex } from '../types/FilterBySex';
+import { useSearchParams } from 'react-router-dom';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const [activeSexFilter, setActiveSexFilter] = useState<FilterBySex>(
+    FilterBySex.ALL,
+  );
+
+  useEffect(() => {
+    if (sex === 'm') {
+      setActiveSexFilter(FilterBySex.MALE);
+    } else if (sex === 'f') {
+      setActiveSexFilter(FilterBySex.FEMALE);
+    } else {
+      setActiveSexFilter(FilterBySex.ALL);
+    }
+  }, [sex]);
+
+  useEffect(() => {
+    getPeople()
+      .then(setPeople)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const visibleList = () => {
```

Comment:
if it's a function it should be named accordingly: `getVisibleList`. Also, in such case, it's better to move it outside of the component and accept all required  data as variables 

Otherwise (better in this case),  you can just use useMemo hook to do the same thing and memorize value

Probably, you are not familiar with useMemo yet. In such case you can apply changes without using this hook (or just read about it if you want 🙂 )